File: youlexuan-manager-web/src/main/webapp/js/controller/brandController.js
//品牌控制层
app.controller('brandController' ,function($scope,$http,$controller,brandService){

    //$controller也是angular提供的一个服务，可以实现伪继承，实际上就是与BaseController共享$scope
    $controller('baseController',{$scope:$scope});//继承父控制器

    //保存:修改JS的save方法
    $scope.save=function(){
        var methodName='add';//方法名称
        if($scope.entity.id!=null){//如果有ID
            methodName='update';//则执行修改方法
        }
        brandService.save(methodName,$scope.entity).success(
            function(response){
                if(response.success){
                    //重新查询
                    $scope.reloadList();//重新加载
                }else{
                    alert(response.message);
                }
            }
        );
    }

    //查询实体
    $scope.findOne=function(id){
        brandService.findOne(id).success(
            function(response){
                $scope.entity= response;
            }
        );
    }

    //批量删除
    $scope.dele=function(){
        //获取选中的复选框
        brandService.dele($scope.selectIds).success(
            function(response){
                if(response.success){
                    $scope.reloadList();//刷新列表
                }else{
                    alert($scope.message);
                }
            }
        );
    }

    //模糊查询
    $scope.searchEntity = {};//定义搜索对象
    //条件查询
    $scope.search = function(page, rows) {
        brandService.search(page,rows,$scope.searchEntity).success(function(response) {
            $scope.paginationConf.totalItems = response.total;//总记录数
            $scope.list = response.rows;//给列表变量赋值
        });
    }

/*
    //分页
    $scope.findPage=function(page,rows){
        $http.get('../brand/findPage.do?page='+page+'&rows='+rows).success(function(response){
                $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }*/
    /* //读取列表数据绑定到表单中：查询所有的品牌列表的方法:
        $scope.findAll=function(){
            //向后台发送请求：
            brandService.findAll().success(function(response){
                    $scope.list=response;
                }
            );
        }*/

    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }

});

File: youlexuan-manager-web/src/main/java/com/youlexuan/manager/controller/BrandController.java
package com.youlexuan.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.youlexuan.pojo.good.Brand;
import com.youlexuan.sellergoods.service.BrandService;
import com.youlexuan.entity.PageResult;
import com.youlexuan.entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;
    /**
     * 返回全部列表
     * @return
     */
    @RequestMapping("/findAll")
    public List<Brand> findAll(){
        return brandService.findAll();
    }

    /**
     * 返回全部列表
     */
    @RequestMapping("/findPage")
    public PageResult findPage(int page,int rows){
        return brandService.findPage(page,rows);
    }

    /**
     * 添加商品品牌
     * @RequestBody：该注解必须加，不加接收不到数据
     * @return
     */
    @RequestMapping("/add")
    public Result insert(@RequestBody Brand brand){
        //return brandService.insert(brand);
        try {
            return  brandService.insert(brand);
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "增加失败");
        }
    }

    /**
     * 修改品牌
     * @param brand
     * @return
     */
    @RequestMapping("/update")
    public Result update(@RequestBody Brand brand){
        try{
            brandService.update(brand);
            return new Result(true,"修改成功");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    /**
     * 获取实体：单个商品品牌信息
     * @param id
     * @return
     */
    @RequestMapping("/findOne")
    public Brand findOne(long id){
        Brand brand = brandService.findOne(id);
        return brand;
    }

    /**
     * 批量删除
     * @param ids
     * @return
     */
    @RequestMapping("/delete")
    public Result delete(Long [] ids){
        try {
            brandService.delete(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "删除失败");
        }
    }
    /**
     * 查询+分页
     * @param brand
     * @param page
     * @param rows
     * @return
     */
    @RequestMapping("/search")
    public PageResult search(@RequestBody Brand brand, int page, int rows  ){
        return brandService.findPage(brand, page, rows);
    }

    /**
     * 列表查询：id和name
     * @return
     */
    @RequestMapping("/selectOptionList")
    public List<Map> selectOptionList(){
        return brandService.selectOptionList();
    }

}
